src/main/java/io/opentracing/contrib/global/GlobalSpanManager.java

```diff
@@ -0,0 +1,87 @@
+package io.opentracing.contrib.global;
+
+import io.opentracing.Span;
+
+import java.io.Closeable;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class manages the globally 'active' {@link Span} in the current Thread.
+ *
+ * @author Sjoerd Talsma
+ */
+public final class GlobalSpanManager {
+
+    private static final Logger LOGGER = Logger.getLogger(GlobalSpanManager.class.getName());
+
+    private static final ThreadLocal<ManagedSpan> ACTIVE_SPAN = new InheritableThreadLocal<ManagedSpan>();
+
+    /**
+     * Static method to return the currently active global {@link Span}.<br>
+     * There is no guarantee that there is an active {@link Span} in all situations.
+     *
+     * @return The currently active global Span, or <code>null</code> if there is no Span currently active.
+     */
+    public static Span activeSpan() {
+        final ManagedSpan active = ACTIVE_SPAN.get();
+        return active != null ? active.span : null; // TODO: Or return no-op Span instead?
+    }
+
+    public static GlobalSpanDeactivator activate(Span span) {
```

Comment:
@bensigelman I tried to keep the idea as simple as possible: return the single Tracer instance if found, or fall back to the no-op tracer otherwise (plus warn if more than one tracer implementation is found).
I like the `MultiplexedTracer` idea, but it brings additional complexity of choosing the 'real' tracer. There is nothing stopping a developer to `register` his/her own multiplexer explicitly. Maybe later we can provide utility code to make this easier?
@bensigelman About the `Runnable` : Something similar was available in the `context-propagation` library which was removed on @wu-sheng's request (to which I agreed). I made the `GlobalSpanManager` explicit in its role of managing the 'active' span. Users of the GlobalTracer need not worry about this class since the delegated SpanBuilder will become 'child-of' the activeSpan by the `start()` override and the returned `Span` will deactivate itself when finished.
_hence all the required delegate implementations_